src/LibraryViewExtension/EventObserver.cs

```diff
@@ -0,0 +1,164 @@
+
+            lock (root) { result = reducer(result, value); }
+
+            OnEventCore(value);
+        }
+
+        /// <summary>
+        /// The core implementation for OnEvent, if throttle is active it will
+        /// update the throttle timer with the due time, so that if no subsequent 
+        /// event is raised within the given throttle due time, the client will 
+        /// be notified.
+        /// </summary>
+        /// <param name="value"></param>
+        protected virtual void OnEventCore(TValue value)
+        {
+            if (throttler != null)
+            {
+                throttler.Change(duetime, TimeSpan.FromMilliseconds(0));
+            }
+            else
+            {
+                Notify(this);
+            }
+        }
+
+        /// <summary>
+        /// Creates an observer that throttle the stream of notification 
+        /// for a given duetime i.e. notification will be fired only if duetime
+        /// is ellapsed from the last event trigger, if the event trigger is
+        /// sooner then the observer will again wait for the given duetime
+        /// to notify the client.
+        /// </summary>
+        /// <param name="duetime">Due time for throttle</param>
+        /// <returns>The modified event Observer which throttle the events</returns>
+        public EventObserver<TValue, TResult> Throttle(TimeSpan duetime)
+        {
+            if (disposed) throw new ObjectDisposedException("EventObserver");
+
+            this.duetime = duetime;
+            this.throttler = new Timer(Notify, this, Timeout.Infinite, Timeout.Infinite);
+            //when throttled, reset the result after notification is fired.
+            accumulate = false;
+            return this;
+        }
+
+        /// <summary>
+        /// Notify the client, could be triggered from throttle timer or 
+        /// called directly by passing this object as state.
+        /// </summary>
+        /// <param name="state"></param>
+        protected void Notify(object state)
+        {
+            if (disposed) throw new ObjectDisposedException("EventObserver");
+
+            TResult r;
+            lock (root)
+            {
+                r = result;
+                if (!accumulate) result = default(TResult);
+            }
+            observer(r);
+        }
+
+        public void Dispose()
+        {
+            if (!disposed)
+            {
+                disposed = true;
+                observer = null;
+                reducer = null;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Implements IDisposable to invoke a given dispose action on Dispose().
+    /// </summary>
+    class AnonymousDisposable : IDisposable
+    {
+        private Action dispose;
+
+        public AnonymousDisposable(Action dispose)
+        {
+            this.dispose = dispose;
+        }
+
+        public void Dispose()
+        {
+            dispose?.Invoke();
+            dispose = null;
```

Comment:
yes, works only with VS2015, BRE self-service builds are failing because of this syntax.